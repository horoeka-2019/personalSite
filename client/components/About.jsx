import React from "react";
import NavInner from "./NavInner";
import Footer from './Footer'
import AboutMeBox from './AboutMeBox'

class About extends React.Component {

  componentDidMount() {
    window.scrollTo(0, 0);
}

  render() {
    return (
      <div>
        <NavInner />
        <div className="container-work-about about-background">
          <header className="title-about-page">
            <img src="../images/aboutfancy.svg" alt="about icon" height="130" />
          </header>
          
          </div>
          <div className='about-row-green-strip'></div>
          <AboutMeBox />
        
        <Footer />
      </div>
      
    );
  }
}

export default About;
