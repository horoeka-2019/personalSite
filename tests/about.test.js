import React from 'react'
import { mount } from 'enzyme'
import About from '../client/components/About'

describe('<About /> component test', () => {
  it('renders title', () => {
    // Arrange
    const expected = 'Jill Glass - Frontend Web Developer'

    // Act
    const wrapper = mount(<About />)
    const actual = wrapper.text()
    // Assert
    expect(actual).toContain(expected)
  })

  it('renders link titles', () => {
    // Arrange
    const expected = 'Github'

    // Act
    const wrapper = mount(<About />)
    const actual = wrapper.text()
    // Assert
    expect(actual).toContain(expected)
  })
 
})