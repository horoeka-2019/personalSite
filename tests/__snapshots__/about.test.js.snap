// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> component test render unchanged from previous snapshot 1`] = `
<About>
  <div>
    <Nav>
      <div>
        <ul
          className="navigation"
        >
          <div
            className="nav-right"
          >
            <HashRouter>
              <Router
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
              >
                <li>
                  <NavLink
                    activeClassName="navbar__link--"
                    className="burger"
                    to="/"
                  >
                    <Link
                      aria-current="page"
                      className="burger navbar__link--"
                      style={Object {}}
                      to={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": null,
                        }
                      }
                    >
                      <LinkAnchor
                        aria-current="page"
                        className="burger navbar__link--"
                        href="#/"
                        navigate={[Function]}
                        style={Object {}}
                      >
                        <a
                          aria-current="page"
                          className="burger navbar__link--"
                          href="#/"
                          onClick={[Function]}
                          style={Object {}}
                        >
                          Home
                        </a>
                      </LinkAnchor>
                    </Link>
                  </NavLink>
                </li>
                <li>
                  <NavLink
                    activeClassName="navbar__link--active"
                    className="burger"
                    to="/about"
                  >
                    <Link
                      aria-current={null}
                      className="burger"
                      to={
                        Object {
                          "hash": "",
                          "pathname": "/about",
                          "search": "",
                          "state": null,
                        }
                      }
                    >
                      <LinkAnchor
                        aria-current={null}
                        className="burger"
                        href="#/about"
                        navigate={[Function]}
                      >
                        <a
                          aria-current={null}
                          className="burger"
                          href="#/about"
                          onClick={[Function]}
                        >
                          About
                        </a>
                      </LinkAnchor>
                    </Link>
                  </NavLink>
                </li>
                <li>
                  <NavLink
                    activeClassName="navbar__link--active"
                    className="burger"
                    to="/work"
                  >
                    <Link
                      aria-current={null}
                      className="burger"
                      to={
                        Object {
                          "hash": "",
                          "pathname": "/work",
                          "search": "",
                          "state": null,
                        }
                      }
                    >
                      <LinkAnchor
                        aria-current={null}
                        className="burger"
                        href="#/work"
                        navigate={[Function]}
                      >
                        <a
                          aria-current={null}
                          className="burger"
                          href="#/work"
                          onClick={[Function]}
                        >
                          Work
                        </a>
                      </LinkAnchor>
                    </Link>
                  </NavLink>
                </li>
              </Router>
            </HashRouter>
          </div>
        </ul>
      </div>
    </Nav>
    <div
      className="container-work-about about-background"
    >
      <header
        className="title"
      >
        <img
          alt="about icon"
          height="130"
          src="../images/aboutfancy.svg"
        />
      </header>
      <h3
        className="entry-title"
      >
        Jill Glass - Frontend Web Developer
      </h3>
      <div
        className="entry-container"
      >
        <div>
          <strong>
            Contact Me
          </strong>
          <div>
            glass.nz@gmail.com
          </div>
        </div>
        <br />
        <div>
          <strong>
            Github
          </strong>
          <div
            className="underline"
          >
            <a
              href="https://github.com/jillGlass"
              target="new"
            >
              https://github.com/jillGlass
            </a>
          </div>
        </div>
        <br />
        <div>
          <strong>
            LinkedIn
          </strong>
          <div
            className="underline"
          >
            <a
              href="https://www.linkedin.com/in/jillglass/"
              target="new"
            >
              https://linkedin.com/in/jillglass/
            </a>
          </div>
        </div>
        <br />
        <div>
          <strong>
            Tech-Stack
          </strong>
          <div>
            React JS, Redux, Semantic UI, CSS, JavaScript, HTML, Adobe XD, Postman, Excel, Dotmailer, Google Analytics, SuperAgent, Testing Suite includes Jest, Enzyme, Nock, SuperTest and Cheerio
          </div>
        </div>
        <br />
        <div>
          <strong>
            Attributes
          </strong>
          <div>
            Disciplined, inquisitive, team-focused, hardworking, respectful, open, organised, understanding, accommodating and social
          </div>
        </div>
        <br />
        <strong>
          My Story
        </strong>
        <div>
          In 2018, with over seven years of marketing experience under my belt, I decided - after taking a break to raise my children - to redirect my career. The first step towards this was to enrol with Girl Code to gain some insight and experience into the dev world and solidify the direction I wanted to take. After Girl Code levels 1 and 2 I enrolled in Enspiral Dev Academy, taking my career full-circle: at 18 I studied Multimedia at SEO in Auckland. 20 years later I am back on that track and am very excited about the future.
        </div>
        <br />
        <div>
          I am a highly focused, organised and collaborative team member who values and fosters a great team culture and who enjoys a fun and hardworking team environment. I enjoy learning from others and I have a curious nature; with my background in marketing, having worked both agency and client side, I understand the necessity of great UX/UI design.
        </div>
        <br />
        <div>
          I also have experience in managing internal stakeholders and clients as well as external clients, building strong, ongoing relationships. I have built web apps using React (including Redux) and Github and have an interest in mobile development - a personal project of mine which I designed and wireframed is now getting off the ground in React Native.
        </div>
      </div>
    </div>
  </div>
</About>
`;
